src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,137 @@
+package duke;
+
+import java.io.File;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+/**
+ * Parses the user command.
+ */
+public class Parser {
+
+    private Scanner sc = new Scanner(System.in);
+    private String filepath = "duke.txt";
+    private Storage storage = new Storage(filepath);
+    private Duke duke = new Duke();
+
+    /**
+     * Constructs a parser.
+     */
+    public Parser() {
+    }
+
+    /**
+     * Interacts with the user based on the command line input.
+     */
+    public void respond() {
+        try {
+            String input = sc.nextLine();
+            TaskList list = new Storage(filepath).load(new File(filepath));
+            String[] descriptions = input.split(" ", 2);
+            String command = descriptions[0];
+            File file = new File(filepath);
+            if (command.equals("mark") || command.equals("unmark")) {
+                if (descriptions.length == 1) {
```

Comment:
I like how you thought about & handled the edge case here! (and the other commands as well!)